### ISSUE TITLE
Style/GuardClause suggests inefficient rewrite of my method

### ISSUE THREAD

**Issue by graywolf** [rubocop-hq/rubocop#5742]
When I have 2 to-level `unless` in my function with each block having more
then 1 line, rubocop complains to use guard clause instead. While that is
possible, it would mean calling the function in the condition twice, slowing
everything done without any benefit. Or leading to some pretty ugly code.

My code looks like this:

```
def foo(f)
  unless File.exist?(f)
    $l.error { 'Does not exist' }
    raise 'Does not exist'
  end
  unless File.directory?(f)
    $l.error { 'Not a dir' }
    raise 'Not a dir'
  end
end
```

rewriting this code using guard clause is possible (but triggers other warnings)

```
def foo(f)
  ( msg = 'Does not exist'; $l.error { msg }; raise msg ) unless File.exist?(f)
  ( msg = 'Not a dir'; $l.error { msg }; raise msg ) unless File.directory?(f)
end
```

but don't tell me it actually improves the code quality.

Other way to rewrite it would be

```
def foo(f)
  $l.error { 'Does not exist' } unless File.exist?(f)
  raise 'Does not exist' unless File.exist?(f)
  $l.error { 'Not a dir' } unless File.directory?(f)
  raise 'Not a dir' unless File.directory?(f)
end
```

But that means accessing filesystem twice instead of just once.

The rule this cop references ( https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals )
talks about nested conditional, I don't believe I have any of those in my code so I'm not entirely sure why it is complaining.

--------

## Expected behavior

To not complain when more then 1 top-level `unless`/`if` is used assuming
the `unless`/`if` in question has more than one line.

## Actual behavior

Rubocop complains even when there are 2 top-level `unless`s used.

## Steps to reproduce the problem

```
[wolf@devel rubocop_test]$ cat test.rb 
class Foo
  def bar
    unless File.exist?('/xxx')
      $l.error { 'File does not exist' }
      raise 'Does not exist'
    end
    unless File.directory?('/xxx')
      $l.error { 'File is not a dir' }
      raise 'Not a dir'
    end
  end
end
[wolf@devel rubocop_test]$ rubocop test.rb 
Inspecting 1 file
C

Offenses:

test.rb:1:1: C: Style/Documentation: Missing top-level class documentation comment.
class Foo
^^^^^
test.rb:3:5: C: Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression.
    unless File.exist?('/xxx')
    ^^^^^^
test.rb:4:7: C: Style/GlobalVars: Do not introduce global variables.
      $l.error { 'File does not exist' }
      ^^

1 file inspected, 3 offenses detected
```

## RuboCop version

```
[wolf@devel rubocop_test]$ rubocop -V
0.54.0 (using Parser 2.5.0.5, running on ruby 2.5.0 x86_64-linux)
```

**Comment by mikegee** [comment]
> But that means accessing filesystem twice instead of just once.

It actually changed 2 filesystem accesses to 4. You can write this method with one filesystem access, btw:

```ruby
class Foo
  def bar
    unless File.stat('/xxx').directory?
      $l.error { 'File is not a dir' }
      raise 'Not a dir'
    end
  rescue Errno::ENOENT
    $l.error { 'File does not exist' }
    raise 'Does not exist'
  end
end
```

Rubocop doesn't complain about a guard clause with this version, but this isn't a general solution to "double unless" style code in your initial example.

**Comment by graywolf** [comment]
I believe this is still an issue in current rubocop:

```
$ gem list | grep rubocop
rubocop (0.59.1)
```

```ruby
# noop
class Foo
  # noop
  def bar
    unless File.exist?('/xxx')
      l.error { 'File does not exist' }
      raise 'Does not exist'
    end
    unless File.directory?('/xxx')
      l.error { 'File is not a dir' }
      raise 'Not a dir'
    end
  end
end
```

```
$ rubocop
Inspecting 1 file
C

Offenses:

test.rb:9:5: C: Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression.
    unless File.directory?('/xxx')
    ^^^^^^

1 file inspected, 1 offense detected
```
